spec/recursive_open_struct_spec.rb

```diff
@@ -96,16 +96,21 @@
 
     context "after a sub-element has been modified" do
       let(:hash) do
-        {
-          :blah => {
-            :blargh => 'Brad'
-          }
-        }
+        { :blah => { :blargh => "Brad" } }
+      end
+      let(:updated_hash) do
+        { :blah => { :blargh => "Janet" } }
       end
+
       subject { RecursiveOpenStruct.new(hash) }
       before(:each) { subject.blah.blargh = "Janet" }
       it "returns a hash that contains those modifications" do
-        subject.to_h.should == { :blah => { :blargh => "Janet" } }
+        subject.to_h.should == updated_hash
+      end
+
+      it "reflects the change at the root node" do
+        table = subject.instance_variable_get(:@table)
+        table.should == updated_hash
```

Comment:
Instead of violating encapsulation, could you test the behavior that is broken? Eg:
```
expect(subject.dup.to_h) .to eq updated_hash
```
Or in Rspec 2.x syntax:
```
subject.dup.to_h.should == updated_hash
```

I think it would also be good to verify that the original has is not modified. Something like:
```
it "does not modify the original input hash" do
  hash[:blah][:blargh].should == 'Brad'
end
```
You could also implicitly test this if you constantized `hash`, but you would need to ensure that all of its sub-elements are constantized as well (eg, the sub-hash and the string).